src/workerd/api/http-test.js

```diff
@@ -209,17 +233,36 @@ export const inspect = {
 
 export const cacheMode = {
   async test() {
+    const cacheEnabled = "cache" in Request.prototype;
     {
       const req = new Request('https://example.org', { });
       assert.strictEqual(req.cache, undefined);
     }
     {
-      const req = new Request('https://example.org', { cache: 'no-store' });
-      assert.strictEqual(req.cache, 'no-store');
+      if(cacheEnabled) {
+        const req = new Request('https://example.org', { cache: 'no-store' });
+        assert.strictEqual(req.cache, 'no-store');
+      } else {
+        assert.throws(() => {
```

Comment:
Shouldn't we also be putting this new behaviour (throwing when `cache` is present) behind the compatibility flag too?

Reply:
When the compat flag is off, we need to be using the old `jsg::Unimplemented` definition, which would make this code irrelevant.

Reply:
we can't exactly be using that dynamically it's part of the type right? Should we just be throwing the same error?

Reply:
Sure you can make it dynamic. Rename `jsg::Unimplemented getCache();` to `jsg::Unimplemented getUnimplementedCache();`, then do something like this in the `JSG_RESOURCE_TYPE` block:

```
if (flags.getCacheOptionEnabled()) {
  JSG_READONLY_PROTOTYPE_PROPERTY(cache, getCache);
} else {
  JSG_READONLY_PROTOTYPE_PROPERTY(cache, getUnimplementedCache);
}
```

So the exported `cache` property in JS hooks up to different implementations depending on the flag.